packages/grid/src/Grid.tsx

```diff
@@ -1896,6 +1896,7 @@ class Grid extends PureComponent<GridProps, GridState> {
       hasVerticalBar,
     } = metrics;
     let { top, left, topOffset, leftOffset } = metrics;
+    const initialMetrics = metrics;
```

Comment:
This is unnecessary - `metrics` will not mutate, and even if it did, you're pointing to the same object so you'd see the same mutations.
```suggestion
```